plugins/out_s3/s3_parquet.c

```diff
@@ -0,0 +1,705 @@
+                              amount_row_group, strlen(amount_row_group));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              " -recordType ", 13);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              ctx->parquet_record_type,
+                              flb_sds_len(ctx->parquet_record_type));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              " -schemaType ", 13);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              ctx->parquet_schema_type,
+                              flb_sds_len(ctx->parquet_schema_type));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              " -schemaFile ", 13);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              ctx->parquet_schema_file,
+                              flb_sds_len(ctx->parquet_schema_file));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              " -output ", 9);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              outfile, strlen(outfile));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf, " ", 1);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf, infile, strlen(infile));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    flb_sds_destroy(amount_page);
+    flb_sds_destroy(amount_row_group);
+
+    return 0;
+
+error:
+    if (amount_page != NULL) {
+        flb_sds_destroy(amount_page);
+    }
+    if (amount_row_group != NULL) {
+        flb_sds_destroy(amount_row_group);
+    }
+
+    return ret;
+}
+
+static int s3_is_dir(const char *dir)
+{
+    int ret;
+    struct stat st;
+
+    if (!dir) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (strlen(dir) == 0) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    ret = stat(dir, &st);
+    if (ret == -1) {
+        return -1;
+    }
+
+    if (st.st_mode & S_IFDIR) {
+        return 0;
+    }
+
+    errno = EINVAL;
+
+    return -1;
+}
+
+static int s3_mkdir(struct flb_s3 *ctx, const char *dir, mode_t mode)
+{
+    struct stat st;
+    char *dup_dir = NULL;
+#ifdef FLB_SYSTEM_WINDOWS
+    char path[PATH_MAX];
+#endif
+#ifdef FLB_SYSTEM_MACOS
+    char *parent_dir = NULL;
+#endif
+
+    int ret;
+
+    if (!stat(dir, &st)) {
+        return 0;
+    }
```

Comment:
Should this check be prior to Line242-247 to only have a single if windows/not block?